slsim/Deflectors/DeflectorTypes/nfw_cluster.py

```diff
@@ -14,18 +15,18 @@ class NFWCluster(DeflectorBase):
     - 'e1_mass': eccentricity of NFW profile
     - 'e2_mass': eccentricity of NFW profile
     - 'z': redshift of deflector
+    - 'subhalos': list of dictionary with subhalo parameters
     """
 
-    def __init__(self, deflector_dict, subhalos_list):
+    def __init__(self, deflector_dict):
         """
 
         :param deflector_dict:  parameters of the cluster halo
         :type deflector_dict: dict
-        :param subhalos_list: list with Deflector instances as cluster subhalos
-        :type subhalos_list: list[Deflector]
         """
+        subhalos_list = deflector_dict["subhalos"]
+        self._subhalos = [EPLSersic(subhalo_dict) for subhalo_dict in subhalos_list]
```

Comment:
this is ok for now, just as a comment: one might want to make optionally different deflector classes possible in future